osu.Game/Scoring/Legacy/LegacyScoreParser.cs

```diff
@@ -57,12 +57,20 @@ public Score Parse(Stream stream)
                 var countKatu = (int)sr.ReadUInt16();
                 var countMiss = (int)sr.ReadUInt16();
 
-                score.ScoreInfo.Statistics[HitResult.Great] = count300;
-                score.ScoreInfo.Statistics[HitResult.Good] = count100;
-                score.ScoreInfo.Statistics[HitResult.Meh] = count50;
-                score.ScoreInfo.Statistics[HitResult.Perfect] = countGeki;
-                score.ScoreInfo.Statistics[HitResult.Ok] = countKatu;
-                score.ScoreInfo.Statistics[HitResult.Miss] = countMiss;
+                var windows = currentRuleset.CreateRulesetContainerWith(workingBeatmap).CreateScoreProcessor().CreateHitWindows();
+
+                if (windows.IsHitResultAllowed(HitResult.Great))
+                    score.ScoreInfo.Statistics[HitResult.Great] = count300;
+                if (windows.IsHitResultAllowed(HitResult.Good))
+                    score.ScoreInfo.Statistics[HitResult.Good] = count100;
+                if (windows.IsHitResultAllowed(HitResult.Meh))
+                    score.ScoreInfo.Statistics[HitResult.Meh] = count50;
+                if (windows.IsHitResultAllowed(HitResult.Perfect))
+                    score.ScoreInfo.Statistics[HitResult.Perfect] = countGeki;
+                if (windows.IsHitResultAllowed(HitResult.Ok))
+                    score.ScoreInfo.Statistics[HitResult.Ok] = countKatu;
+                if (windows.IsHitResultAllowed(HitResult.Miss))
+                    score.ScoreInfo.Statistics[HitResult.Miss] = countMiss;
```

Comment:
This is still not done correctly, it needs to be cased per-mode. E.g. Catch uses 300 + gekis for our "PERFECT".